File: guess_number/guess_number.erl
-module(guess_number).
-export([guess_number/0]).

guess_number(Solution, Num) ->
    io:format("Guess ~w: ", [Num]),
    {ok, [Guess]} = io:fread("", "~d"),
    guess_number(Guess, Solution, Num).

guess_number(_, Solution, Num) when Num >= 5 ->
    io:format("Haha, I won! The number was ~w~n", [Solution]),
    init:stop();
guess_number(Attempt, Solution, Num) when Attempt > Solution ->
    io:fwrite("Too high! Try again!\n"),
    guess_number(Solution, Num+1);
guess_number(Attempt, Solution, Num) when Attempt < Solution ->
    io:fwrite("Too low! Try again!\n"),
    guess_number(Solution, Num+1);
guess_number(_, _, _) ->
    io:fwrite("Correct! You have won!\n"),
    init:stop().

guess_number() ->
    random:seed(now()),
    Solution = random:uniform(100),
    
    io:fwrite("Guess-a-number game!\n"),
    io:fwrite("I am thinking of a number between 1 and 100.\n"),
    io:fwrite("You have 5 tries to guess it correctly or I win.\n"),
    io:fwrite("What's your guess?\n"),
    guess_number(Solution, 1).

    
% TAIL INFO:
% Name: Guess Number
% Language: Erlang
% Compile: erlc guess_number.erl
% State: Done
%
% Play guess-a-number game
%
%
% Example: erl -noshell -run guess_number guess_number
%

File: guess_number/guess_number.hs
module Main where

import qualified System.Random as R
import qualified System.IO as IO

guessNumber :: (Integer, Integer) -> IO ()
guessNumber (solution, num)
  | num == 6 = putStrLn $ "Haha, I won! The number was " ++ show solution
  | otherwise = do
    putStr $ "Guess " ++ show num ++ ": "
    IO.hFlush IO.stdout
    guess <- getLine
    case (read guess) `compare` solution of
      LT -> do
        putStrLn "Too low! Try again!"
        guessNumber(solution, succ num)
      GT -> do
        putStrLn "Too high! Try again!"
        guessNumber(solution, succ num)
      EQ -> do
        putStrLn "Correct! You have won!"

main = do
  solution <- R.randomRIO(1, 100) :: IO Integer
  putStrLn "Guess-a-number game!"
  putStrLn "I am thinking of a number between 1 and 100"
  putStrLn "You have 5 tries to guess it correctly or I win"
  putStrLn "What's your guess?"
  guessNumber(solution, 1)
  
-- TAIL INFO:
-- Name: Guess number
-- Language: Haskell
-- Compile: ghc --make guess_number.hs
-- State: Done
--
-- Play guess-a-number game
--
-- Example: ./guess_number
